spec/dry/types/constrained_spec.rb

```diff
@@ -51,6 +51,20 @@
     end
   end
 
+  describe '#to_s' do
+    subject(:type) do
+      Dry::Types['strict.string'].constrained(size: 3..12)
+    end
+
+    it 'returns string representation of the type' do
+      expect(type.to_s).
+        to eql(
+            "#<Dry::Types[Constrained<Definition<String> "\
+            "rule=[type?(String) AND size?(3..12)]>]>"
```

Comment:
I love the " AND " here 👍

Reply:
Actually, this comes straight from dry-logic, I did nothing to represent it this :)

Reply:
@timriley oh you didn't know that dry logic rules are coercible to nice string representation? Check this out:

``` ruby
schema = Dry::Schema.Params do
  required(:email).filled(:string)
  required(:age).filled(:integer, gt?: 18)
  required(:phone) { array? & min_size?(1) & each(:str?, min_size?: 6) | str? & min_size?(6) }
end

schema.rules.values.map(&:to_s)
# [
#   "key?(:email) AND key[email](str? AND filled?)",
#   "key?(:age) AND key[age](int? AND filled? AND gt?(18))", "key?(:phone) AND key[phone](array? AND min_size?(1) AND each(str? AND min_size?(6)) OR str? AND min_size?(6))"
# ]
```

Reply:
@solnic that is super nice :)